egs2/TEMPLATE/spk1/spk.sh

```diff
@@ -72,6 +72,27 @@ inference_config=conf/decode.yaml   # Inference configuration
 inference_model=valid.eer.best.pth  # Inference model weight file
 score_norm=false      # Apply score normalization in inference.
 qmf_func=false        # Apply quality measurement based calibration in inference.
+cos_sim=false       # Use cosine similarity for scoring
+
+# Speaker Task related
+multi_task=false      # Apply multi-task learning in speaker recognition
+sasv_task=false       # Apply speaker anti-spoofing task in speaker recognition
+spf_args=            # Parameters for spoofing
+
+# Embedding averaging related
+embed_avg=false    # Apply embedding averaging in inference
+embed_avg_args=    # Arguments for embedding averaging
+
+# Use pseudomos scores
+pseudomos=false   # Use pseudomos scores for training
+train_utt2pmos=   # absolue path to utt2pmos file, required if pseudomos is true
+valid_utt2pmos=   # absolue path to utt2pmos file, required if pseudomos is true
+
+# use precomputed feats
+precomputed_feats=false        # Use precomputed features for training
+
+# Label related
+no_labels=false    # No labels for inference
```

Comment:
I'm not sure about how several variables would be used. 
Several of them also seem to overlap with existing variables. 
Can you let me know what each variable does?

Reply:
after all these changes, I'm starting to think that maybe adding an SASV task would be better. How does this sound to you?

Reply:
Ok let me try:
multi_task - require two sets of labels used along with sasv_task for multi task learning
sasv_task - defines that the secondary task is spoofing
spf_args - collects a bunch of args similar to spk_args
embed_avg - embedding averaging for enrollment utterances
embed_avg_args - picks up the extra args for embedding averaging (trial3.scp and trial4.scp)
pseudomos - if true then approximating pseudomos scores will be an secondary objective
train_utt2pmos and valid_utt2pmos - has pmos labels for a given utt
precomputed_feats - allows precomputed features to be passed as .npy arrays
no_labels - this is because by default ESPnet SPK requires trial_label but for ASVspoof5 test no such labels were available. I could have missed something because I struggled to find any other workaround for test sets where you only want to compute scores, since labels aren't present.

I see. Could you kindly let me know why a new task would be better?
I'm okay with it, if a new task will result in the best outcomes... then I will definitely switch to that.